File: Domain/Products/Product.cs

namespace Domain.Products
{
    public class Product
    {
        public Guid Id { get; }
        public String Name { get; }
        public ProductCategory Category { get; }
        public Guid GroupId { get; }
        public String Description { get; }
        public Boolean IsSale { get; }
        public Decimal Price { get; }
        public Boolean IsRemoved { get; }

        public Product(Guid id, String name, ProductCategory category, Guid groupId, String description, Boolean sale, Decimal price, Boolean isRemoved)
        {
            Id = id;
            Name = name;
            Category = category;
            GroupId = groupId;
            Description = description;
            IsSale = sale;
            Price = price;
            IsRemoved = isRemoved;
        }
         
    }
}

File: WebApp/ClientApp/src/domain/products/models/product.ts
import { ProductCategory } from "./productCategory";

export class Product {
     id: string
     name: string
     category: ProductCategory
     groupid: string
     description: string
     issale: boolean
     price: number

     constructor(id:string, name:string, category: ProductCategory, groupid: string, description: string, issale: boolean, price: number) {
        this.id = id
        this.name = name
        this.category = category
        this.groupid = groupid
        this.description = description
        this.issale = issale
        this.price = price       
     }
}


export function mapToProduct(data: any): Product{
    return new Product(data.id, data.name, data.category, data.groupId, data.description, data.isSale, data.price)
}

export function mapToProducts(data: any[]): Product[]{
    return data.map(d => mapToProduct(d))
}